src/main/java/frc/robot/subsystems/UpBeat.java

```diff
@@ -54,38 +57,61 @@ public UpBeat() {
         bottomMotor.burnFlash();
     }
 
+    private final static class Constants{
+        private static final double stop = 0.00;
+        private static final double reverse = -1000.00;
+        private static final double shoot = 4000.00;
+        private static final double amp = 1000.00;
+    }
+
+    @AutoLogOutput(key = "upBeat/speed")
+    private double getSpeed() {
+        return bottomMotor.getEncoder().getVelocity();
+    }
+
+    @AutoLogOutput(key = "upBeat/error")
+    private double getError(){
+        return setPoint-getSpeed();
+    }
+
+    private boolean atSpeed(){
+        if(getError()<-100 && getError()>100){
+        return true;
+        }
+        return false;
+    }
+
+    private void setSpeed(){
+        topPid.setReference((setPoint*1.2), ControlType.kVelocity);
+        bottomPid.setReference(setPoint, ControlType.kVelocity);
+    }
+
+    private Command speedCommand(double speed){
+        setPoint = speed;
```

Comment:
This will set the speed on construction of the command, instead we'll want to put this in the lambda below like so: `return run(() -> setpoint = speed)...`